src/components/atoms/InputField/InputField.tsx

```diff
@@ -0,0 +1,58 @@
+import React, { forwardRef } from "react";
+import classNames from "classnames";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+import { IconProp } from "@fortawesome/fontawesome-svg-core";
+import { isTruthy } from "utils/types";
+import "./InputField.scss";
+
+export interface InputFieldProps extends React.HTMLProps<HTMLInputElement> {
+  iconStart?: IconProp | JSX.Element | false;
+  iconEnd?: IconProp | JSX.Element | false;
+}
+
+const iconClassNames = (position: string) =>
+  classNames("input-field__icon", `input-field__icon--${position}`);
+
+const isJsxElement = (
+  component: IconProp | JSX.Element
+): component is JSX.Element => {
+  return isTruthy(component?.hasOwnProperty("props"));
+};
+
+const renderIcon = (
+  icon: IconProp | JSX.Element,
+  position: string
```

Comment:
Since you are only using `position` as a way to generate a class, wouldn't it be better just to pass a classname?

Reply:
updated: https://github.com/sparkletown/sparkle/pull/1306/commits/2778ea8bf7b45b4016fea94b850bae5e0a449c92

Reply:
@mike-lvov Why do you say that? The fact that it’s used in the class names feels like an irrelevant internal detail that shouldn’t need to be exposed in the ‘public API’ of this component.